src/zkevm_specs/evm_circuit/execution/create.py

```diff
@@ -130,71 +81,181 @@ def create(instruction: Instruction):
         all_but_one_64th_gas,
     )
 
-    # copy init_code from memory to bytecode
-    copy_rwc_inc, _ = instruction.copy_lookup(
-        instruction.curr.call_id,  # src_id
-        CopyDataTypeTag.Memory,  # src_type
-        instruction.next.code_hash,  # dst_id
-        CopyDataTypeTag.Bytecode,  # dst_type
-        offset,  # src_addr
-        offset + size,  # src_addr_boundary
-        FQ(0),  # dst_addr
-        size,  # length
-        instruction.curr.rw_counter + instruction.rw_counter_offset,
+    ### Do stack depth, nonce and balance pre-check
+    # ErrDepth constraint
+    is_error_depth, _ = instruction.compare(FQ(CALL_CREATE_DEPTH), depth, N_BYTES_STACK)
+    # ErrNonceUintOverflow constraint
+    is_nonce_overflow, _ = instruction.compare(nonce, nonce_prev, N_BYTES_U64)
+    # ErrInsufficientBalance constraint
+    is_insufficient_balance, _ = instruction.compare_word(Word(balance_prev.expr().n), value_word)
+
+    # pass the pre-check if none of above errors happen
+    is_precheck_ok = (
+        is_error_depth == FQ(0) and is_nonce_overflow == FQ(0) and is_insufficient_balance == FQ(0)
     )
-    instruction.rw_counter_offset += int(copy_rwc_inc)
-
-    # CREATE:  3 pops and 1 push, stack delta = 2
-    # CREATE2: 4 pops and 1 push, stack delta = 3
-    stack_pointer_delta = 2 + is_create2
-    # Save caller's call state
-    for field_tag, expected_value in [
-        (CallContextFieldTag.ProgramCounter, instruction.curr.program_counter + 1),
-        (
-            CallContextFieldTag.StackPointer,
-            instruction.curr.stack_pointer + stack_pointer_delta,
-        ),
-        (CallContextFieldTag.GasLeft, gas_left - gas_cost - callee_gas_left),
-        (CallContextFieldTag.MemorySize, next_memory_size),
-        (
-            CallContextFieldTag.ReversibleWriteCounter,
-            instruction.curr.reversible_write_counter + 1,
-        ),
-    ]:
+
+    # error cases, should end this call and return
+    # is_static should be false
+    if not is_precheck_ok or is_static.expr() == FQ(1):
```

Comment:
I think we should just constrain that `is_static == 0`.

Reply:
fixed in [22b6eff](https://github.com/privacy-scaling-explorations/zkevm-specs/pull/416/commits/22b6effbb5dc11a274a9382e089bdc7d439bcb9e)

Reply:
I think we can remove `or is_static.expr() == FQ(1)` here

Reply:
fixed in [048ab74](https://github.com/privacy-scaling-explorations/zkevm-specs/pull/416/commits/048ab74e8c19e9d3d997c416f42221c68afe31a1)